pkg/systemd/services/services.jsx

```diff
@@ -0,0 +1,739 @@
+            stateDropdownIsExpanded: false,
+            currentTypeFilter: null,
+            currentTextFilter: '',
+
+            unit_by_path: {},
+            loadingUnits: true,
+            privileged: true,
+            path: cockpit.location.path,
+            alerts: [],
+            tabErrors: {},
+        };
+        /* Functions for controlling the toolbar's components */
+        this.onClearAllFilters = this.onClearAllFilters.bind(this);
+        this.onTypeDropdownSelect = this.onTypeDropdownSelect.bind(this);
+        this.onInputChange = this.onInputChange.bind(this);
+
+        /* Function for manipulating with the API results and store the units in the React state */
+        this.processFailedUnits = this.processFailedUnits.bind(this);
+        this.getUnitByPath = this.getUnitByPath.bind(this);
+        this.updateProperties = this.updateProperties.bind(this);
+        this.addTimerProperties = this.addTimerProperties.bind(this);
+        this.updateComputedProperties = this.updateComputedProperties.bind(this);
+        this.compareUnits = this.compareUnits.bind(this);
+
+        this.permission = cockpit.permission({ admin: true });
+        this.onPermissionChanged = this.onPermissionChanged.bind(this);
+
+        this.seenPaths = new Set();
+        this.path_by_id = {};
+        this.operationInProgress = {};
+
+        this.on_navigate = this.on_navigate.bind(this);
+        this.addAlert = this.addAlert.bind(this);
+        this.removeAlert = this.removeAlert.bind(this);
+    }
+
+    componentDidMount() {
+        /* Listen for permission changes for "Create Timer" button */
+        this.permission.addEventListener("changed", this.onPermissionChanged);
+        this.onPermissionChanged();
+
+        cockpit.addEventListener("locationchanged", this.on_navigate);
+        this.on_navigate();
+
+        /* Prepare the "Create Timer" dialog - TODO: this needs to be rewritten in React */
+        timerDialogSetup();
+
+        systemd_manager.wait(() => {
+            if (systemd_manager.valid) {
+                this.systemd_subscription = systemd_manager.Subscribe()
+                        .catch(error => {
+                            if (error.name != "org.freedesktop.systemd1.AlreadySubscribed" &&
+                            error.name != "org.freedesktop.DBus.Error.FileExists")
+                                console.warn("Subscribing to systemd signals failed", error);
+                        });
+                this.listUnits();
+            } else {
+                console.warn("Connecting to systemd failed");
+            }
+        });
+
+        /* Start listening to signals for updates - when in the middle of reload mute all signals
+         * - We don't need to listen to 'UnitFilesChanged' signal since everytime we
+         *   perform some file operation we do call Reload which issues 'Reload' signal
+         * - JobNew is also useless, JobRemoved is enough since it comes in pair with JobNew
```

Comment:
The reason I approached it like this, is that all operations that we support from the UI that trigger such a signal are mostly instant. (start, stop, reset unit)
So getting two updates for the same operation instead of one seems a bit redundant. Although the properties probably have different values between the two events (JobRemoved, JobNew)

But I guess in edgy scenarios the call might get stuck and you need to see the intermediate state as well.
I am just not convinced that we should pay the LoadUnit getall properties extra time just for these edgy cases.